compiler/src/test/scala/org/scalajs/nscplugin/test/StaticForwardersASTTest.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.nscplugin.test
+
+import org.scalajs.nscplugin.test.util._
+
+import org.scalajs.ir.Names._
+import org.scalajs.ir.Trees._
+import org.scalajs.ir.Types._
+
+import org.junit.Assert._
+import org.junit.Test
+
+// scalastyle:off line.size.limit
+
+class StaticForwardersASTTest extends JSASTTest  {
+
+  @Test
+  def emitStaticForwardersInExistingClass(): Unit = {
+    val classDef = """
+    import scala.scalajs.js, js.annotation._
+
+    class Foo(val y: Int = 10)
+
+    object Foo {
+      def bar(x: Int = 5): Int = x + 1
+
+      @js.native
+      @JSGlobal("foobar")
+      def foobar(x: Int = 5): Int = js.native
+    }
+    """.extractOne("class Foo") {
+      case cd: ClassDef if cd.name.name == ClassName("Foo") => cd
+    }
+
+    val staticMethodNames = classDef.memberDefs.collect {
+      case MethodDef(flags, MethodIdent(name), _, _, _, _) if flags.namespace.isStatic =>
+        name
+    }.sortBy(_.simpleName)
```

Comment:
Use a `Set` instead for unordered comparison?

Reply:
I prefer the sorted comparison because, when it fails, it is easier to see what differs between the two lists.